# Project 1: Deaths by tuberculosis

by Michel Wermelinger, 14 July 2015, with minor edits on 5 April 2016

additional edits by George Kaimakis, 10 May 2017

This is the project notebook for Week 1 of The Open University's [_Learn to code for Data Analysis_](http://futurelearn.com/courses/learn-to-code) course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2013 in each of the BRICS (Brazil, Russia, India, China, South Africa) and Portuguese-speaking countries. 

The data was taken in July 2015 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.593?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file. This data file must be in a location path accessable from this notebook's file location (see below).

In [1]:
import warnings
warnings.simplefilter('ignore', FutureWarning)

from pandas import *
pandas.set_option('display.max_rows', 30)   # alter this value (30) in order to display more or less of the table
data = read_excel('../Datasets/Datasets_From_Course/WHO POP TB all.xls') # relative path to the datafile location
data

Country  Population (1000s)  \
0                                          Afghanistan               30552   
1                                              Albania                3173   
2                                              Algeria               39208   
3                                              Andorra                  79   
4                                               Angola               21472   
5                                  Antigua and Barbuda                  90   
6                                            Argentina               41446   
7                                              Armenia                2977   
8                                            Australia               23343   
9                                              Austria                8495   
10                                          Azerbaijan                9413   
11                                             Bahamas                 377   
12                                             Bahrain                1332   
13                                          Bangladesh              156595   
14                                            Barbados                 285   
..                                                 ...                 ...   
179                                             Tuvalu                  10   
180                                             Uganda               37579   
181                                            Ukraine               45239   
182                               United Arab Emirates                9346   
183  United Kingdom of Great Britain and Northern I...               63136   
184                        United Republic of Tanzania               49253   
185                           United States of America              320051   
186                                            Uruguay                3407   
187                                         Uzbekistan               28934   
188                                            Vanuatu                 253   
189                 Venezuela (Bolivarian Republic of)               30405   
190                                           Viet Nam               91680   
191                                              Yemen               24407   
192                                             Zambia               14539   
193                                           Zimbabwe               14150   

     TB deaths  
0     13000.00  
1        20.00  
2      5100.00  
3         0.26  
4      6900.00  
5         1.20  
6       570.00  
7       170.00  
8        45.00  
9        29.00  
10      360.00  
11        1.80  
12        9.60  
13    80000.00  
14        2.00  
..         ...  
179       2.80  
180    4100.00  
181    6600.00  
182      64.00  
183     340.00  
184    6000.00  
185     490.00  
186      40.00  
187    2200.00  
188      16.00  
189     480.00  
190   17000.00  
191     990.00  
192    3600.00  
193    5700.00  

[194 rows x 3 columns]

## The range of the problem

The column of interest is the last one.

In [2]:
tbColumn = data['TB deaths']

The total number of deaths in 2013 is:

In [3]:
tbColumn.sum()  # the 'tbColumn' variable is evaluated with several 'methods' - sum(), max(), etc

1072677.97

The largest and smallest number of deaths in a single country are:

In [4]:
tbColumn.max()

240000.0

In [5]:
tbColumn.min()

0.0

From less than 20 to almost a quarter of a million deaths is a huge range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [6]:
tbColumn.mean()

5529.267886597938

In [7]:
tbColumn.median()

315.0

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2013, pushing the value of the mean up.

## Total deaths per country due to TB

To see the most affected countries, the table is sorted in decending order by the last column, which puts those countries in the first rows.

In [8]:
data.sort('TB deaths', ascending=False)    # the 'keyword' property results in a descending sort - highest at the top

Country  Population (1000s)  TB deaths
77                              India             1252140  240000.00
124                           Nigeria              173615  160000.00
13                         Bangladesh              156595   80000.00
78                          Indonesia              249866   64000.00
128                          Pakistan              182143   49000.00
47   Democratic Republic of the Congo               67514   46000.00
36                              China             1393337   41000.00
58                           Ethiopia               94101   30000.00
134                       Philippines               98394   27000.00
116                           Myanmar               53259   26000.00
159                      South Africa               52776   25000.00
115                        Mozambique               25834   18000.00
190                          Viet Nam               91680   17000.00
141                Russian Federation              142834   17000.00
0                         Afghanistan               30552   13000.00
..                                ...                 ...        ...
11                            Bahamas                 377       1.80
143             Saint Kitts and Nevis                  54       1.60
105                             Malta                 429       1.50
152                        Seychelles                  93       1.40
113                        Montenegro                 621       1.20
5                 Antigua and Barbuda                  90       1.20
68                            Grenada                 106       1.10
76                            Iceland                 330       0.93
118                             Nauru                  10       0.67
40                       Cook Islands                  21       0.41
129                             Palau                  21       0.36
3                             Andorra                  79       0.26
111                            Monaco                  38       0.03
125                              Niue                   1       0.01
147                        San Marino                  31       0.00

[194 rows x 3 columns]

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

The death rate per 100,000 inhabitants is show in decending order in the last column, which puts the countries with the highest death rates in the first rows.

In [9]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100 / populationColumn # a new column of data is created from the evaluation
data.sort('TB deaths (per 100,000)', ascending=False)   # the table is sorted in descending order base on the new column

Country  Population (1000s)  TB deaths  \
49                           Djibouti                 873     870.00   
124                           Nigeria              173615  160000.00   
165                         Swaziland                1250    1100.00   
172                       Timor-Leste                1133     990.00   
158                           Somalia               10496    7700.00   
71                      Guinea-Bissau                1704    1200.00   
115                        Mozambique               25834   18000.00   
47   Democratic Republic of the Congo               67514   46000.00   
30                           Cambodia               15135   10000.00   
117                           Namibia                2303    1300.00   
62                              Gabon                1672     910.00   
92   Lao People's Democratic Republic                6770    3600.00   
100                        Madagascar               22925   12000.00   
13                         Bangladesh              156595   80000.00   
96                            Liberia                4294    2100.00   
..                                ...                 ...        ...   
167                       Switzerland                8078      17.00   
82                             Israel                7733      16.00   
44                             Cyprus                1141       2.30   
113                        Montenegro                 621       1.20   
8                           Australia               23343      45.00   
32                             Canada               35182      62.00   
16                            Belgium               11104      18.00   
185          United States of America              320051     490.00   
121                       New Zealand                4506       6.30   
166                            Sweden                9571      13.00   
137                             Qatar                2169       2.70   
120                       Netherlands               16759      20.00   
126                            Norway                5043       4.40   
111                            Monaco                  38       0.03   
147                        San Marino                  31       0.00   

     TB deaths (per 100,000)  
49                 99.656357  
124                92.157936  
165                88.000000  
172                87.378641  
158                73.361280  
71                 70.422535  
115                69.675621  
47                 68.134017  
30                 66.072019  
117                56.448111  
62                 54.425837  
92                 53.175775  
100                52.344602  
13                 51.087199  
96                 48.905449  
..                       ...  
167                 0.210448  
82                  0.206905  
44                  0.201578  
113                 0.193237  
8                   0.192777  
32                  0.176226  
16                  0.162104  
185                 0.153101  
121                 0.139814  
166                 0.135827  
137                 0.124481  
120                 0.119339  
126                 0.087250  
111                 0.078947  
147                 0.000000  

[194 rows x 4 columns]

## Conclusions

The entire world population had a total of about 1.07 million deaths due to TB in 2013. The median shows that half of these coutries had less than 315 deaths. The much higher mean (over 5,500) indicates that some countries had a very high number. The least affected were San Marino, Niuo and Monaco, with less than 0.1 deaths, and the most affected were India and Nigeria with 240 thousand and 160 thousand deaths in a single year respectively. However, taking the population size into account, the least affected were San Marino, Monaco and Norway with less than 0.1 deaths per 100 thousand inhabitants, and the most affected were Djibouti and Nigeria with over 92 deaths per 100,000 inhabitants.

One should not forget that most values are estimates. Nevertheless, they convey the message that TB is still a major cause of fatalities, and that there is a huge disparity between countries, with several countries being highly affected.